src/NodaTime.Web/Markdown/developer/website.md

```diff
@@ -0,0 +1,89 @@
+@Title="The nodatime.org website"
+
+## Overview
+
+[The nodatime.org website](http://nodatime.org/) is currently served from an
+ASP.NET Core web application hosted by Azure. The canonical source for the
+website is the [nodatime.org GitHub
+repository](https://github.com/nodatime/nodatime.org); changes to this
+repository are automatically deployed to the website (via a webhook that
+pings Azure to perform a pull/redeploy).
+
+However, the nodatime.org repository is typically _not_ edited directly,
+except in an emergency. Instead, changes are made in the nodatime
+repository, and then the nodatime.org repository is generated from that.
+
+The source for most of the website is the [NodaTime.Web
+project](https://github.com/nodatime/nodatime/tree/master/src/NodaTime.Web).
+This contains both the web application and the website content,
+particularly:
+
+- [`Markdown/`](https://github.com/nodatime/nodatime/tree/master/src/NodaTime.Web/Markdown)
+  containing the Markdown content used to generate the user and developer
+  guides.
+- [`wwwroot/`](https://github.com/nodatime/nodatime/tree/master/src/NodaTime.Web/wwwroot)
+  containing static content for the site.
+- [`Views/`](https://github.com/nodatime/nodatime/tree/master/src/NodaTime.Web/Views)
+  containing the programmatic content, for e.g. the home page and downloads
+  page.
+
+The [downloads](/downloads/) and [TZDB NZD files](/tzdb/) are stored on
+Google Cloud Storage, under the `https://storage.googleapis.com/nodatime/`
+bucket. The downloads page links directly to those files, but the NZD files
+are fetched and re-served via the nodatime.org web application (mostly
+because we do not want to serve redirects or non-nodatime.org URLs to
+clients).
```

Comment:
Do we want to mention that the NZD files are cached after fetching? Probably not vital. It does mean that if we ever need to modify an NZD file instead of releasing a new one, we'd need to restart the web app.